### ISSUE TITLE
F-droid in sources filter

### ISSUE THREAD

**Issue by Logmytech** [MartinStyk/AndroidApkAnalyzer#11]
F-Droid apps show up as sideloaded apps and there isn't an easy way to know if the app is sideloaded.

Since F-Droid is the only source for FOSS exclusive apps, it might be nice to add a Filter for F-Droid and filter the F-Droid signed apps.

**Comment by MartinStyk** [comment]
Hi, thank you for opening the issue. I will take a look and see what can be done here.

**Comment by MartinStyk** [comment]
Hi @Logmytech , unfortunately there is no way to distinguish F-droid apps (at least I am not aware of any).
Only thing that can be used to find the source of application is a package that installed it. This package differs between app stores. However, F-droid downloads an application and shows prompt to install it using standard package installer. Because of that, it is impossible to distinguish these apps.

**Comment by Logmytech** [comment]
But the signature of APK Analyzer shows F-droid.

So, F-Droid actually signs these APKS. So we can check the signatures in the APK, maybe?

**Comment by MartinStyk** [comment]
Yeah, it signs it. But in that case, I would need all keys used for signing apps in F-droid market. I can not rely on name in signature, becasue anyone can create signing key with name F-droid. I would need to match against the public key. 
I do not think it is a good idea to do it, because it will not be reliable...
wdyt?

**Comment by IzzySoft** [comment]
@MartinStyk you can tell the installation source by the corresponding attribute (`-i` parameter to `pm`); in the package dump, the field is called `installerPackageName`. Playstore has two different "sources" here, FDroid just one (`org.fdroid.fdroid` if I remember correctly), Aptoide has its own as well (as will all other market apps, I suspect). Just create a dump and grep for `installerPackageName`, sort, and `uniq` :wink:

**Comment by MartinStyk** [comment]
@Logmytech @IzzySoft,
I get the installation source using the {{PackageManager}}'s method {{getInstallerPackageName}} [1]. It is basically the same as described in @IzzySoft's comment.
However, when I test it, for F-droid apps I always get installer package {{com.google.android.packageinstaller}}, which is default Android installer.

I suppose it is because F-Droid app downloads an apk file, but let default android installer to install the package. 

Am I missing something here?
Thank you for your help 👍 

[1]https://github.com/MartinStyk/AndroidApkAnalyzer/blob/master/app/src/main/java/sk/styk/martin/apkanalyzer/model/detail/AppSource.java#L34

**Comment by IzzySoft** [comment]
What I do in my tool [Adebar](https://github.com/IzzySoft/Adebar) is parsing the package list returned by `dumpsys package` (starting at `^Packages:` and stopping at `^Shared users:`). And *Adebar* reports the correct installer. The Android installer certainly is invoked the same way `pm` is (a la `pm install -i <installer_package_name> …`).

I'm no Android dev, so I don't know any corresponding Java APIs, sorry. If you want to cross-check with my Shell code, see the function `getAppDetails()` in `lib/packagedata.lib`.
